src/anyio/_backends/_asyncio.py

```diff
@@ -1536,11 +1595,16 @@ def statistics(self) -> CapacityLimiterStatistics:
                                          tuple(self._borrowers), len(self._wait_queue))
 
 
-def current_default_thread_limiter():
-    return _default_thread_limiter
+_default_thread_limiter: RunVar[CapacityLimiter] = RunVar('_default_thread_limiter')
 
 
-_default_thread_limiter = CapacityLimiter(40)
+def current_default_thread_limiter():
+    try:
+        return _default_thread_limiter.get()
+    except LookupError:
+        limiter = CapacityLimiter(40)
```

Comment:
@agronholm Does this means that the default behavior will allow to spawn at most 40 threads? Did you consider `os.cpu_count()` that is what is used in `asyncio.to_thread`?

ref: https://github.com/python/cpython/blob/main/Lib/concurrent/futures/thread.py#L142

Reply:
This is in line with trio.to_thread which also allows a maximum of 40 threads by default.